src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,312 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Observable } from "rxjs/Observable";
+import assert from "../../utils/assert";
+import request from "../../utils/request";
+import BoxPatcher from "./isobmff_patcher";
+
+import { ISegmentPrivateInfos } from "../../manifest/representation_index/interfaces";
+import { generateManifest } from "../../parsers/manifest/metaplaylist/index";
+import { IKeySystem } from "../../parsers/manifest/types";
+import {
+    CustomSegmentLoader,
+    ILoaderObservable,
+    ILoaderResponse,
+    ImageParserObservable,
+    IManifestLoaderArguments,
+    IManifestParserArguments,
+    IManifestParserObservable,
+    IMetaTransportPipelines,
+    ISegmentLoaderArguments,
+    ISegmentParserArguments,
+    SegmentParserObservable,
+  } from "../types";
+
+import DASHTransport from "../dash";
+import SmoothTransport from "../smooth";
+
+interface IParserOptions {
+  segmentLoader? : CustomSegmentLoader;
+  suggestedPresentationDelay? : number;
+  referenceDateTime? : number;
+  minRepresentationBitrate? : number;
+  keySystems? : (hex? : Uint8Array) => IKeySystem[];
+}
+
+export interface IMetaManifestInfo {
+    manifests: Array<{
+      manifest: Document;
+      url: string;
+      startTime: number;
+      endTime: number;
+      transport: "dash"|"smooth";
+    }>;
+}
+
+type transportTypes = "dash"|"smooth";
+
+const transportTypes: transportTypes[] = ["dash", "smooth"];
+
+/**
+ * Parse playlist string to JSON.
+ * Returns an array of contents.
+ * @param {string} data
+ */
+function loadMetaPlaylistData(data: string): Array<{
+    url: string;
+    startTime: number;
+    endTime: number;
+    transport: "dash"|"smooth";
+  }> {
+  const parsedMetaPlaylist = JSON.parse(data);
+  const { contents } = parsedMetaPlaylist;
+  assert(contents);
+
+  return contents;
+}
+
+/**
+ * Get transport type from segment's privateInfos
+ * @param {Object} privateInfos
+ */
+function getTypeFromPrivateInfos(privateInfos: ISegmentPrivateInfos): transportTypes {
+  const transportType = privateInfos.manifestType;
+
+  if (!transportType) {
+    throw new Error("Undefined transport for content for metaplaylist.");
+  }
+
+  return transportType;
+}
+
+export default function(
+    options: IParserOptions = {}
+): IMetaTransportPipelines{
+
+    const transports = {
+      dash: DASHTransport(options),
```

Comment:
What if ``FEATURES.DASH`` is deactivated?

Note: We can just say we still include it if METADASH is included. But that must be clear